Create Blog “set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud”

GuopingJia opened the pull request:
Automatically generated by Netlify CMS

choukden commented on content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:
````diff
@@ -0,0 +1,324 @@
+    silence-period: 3
+    message: Successful rate of service {name} is lower than 80% in 2 minutes of last 10 minutes
+```
+W﻿e can go to the frontend app by clicking the service `agent::app` from SkyWalking UI Service page. From the below service `agent::app` overview page, it shows **Success Rate 66.66%**. 
````
You can go to the....

choukden reviewed (changes requested):
Hi Guoping,

Thank you very much for this blog post of your great series.

Here are some suggestions for edits.

Dale will go through the editorial review in the coming day or so, So expect to get additional comments and suggestions for edit in the coming days.

best regards,
Denis

choukden commented on content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:
```diff
@@ -0,0 +1,324 @@
+
+This blog post took the Apache SkyWalking as the application performance monitoring (APM) tool and showed the detailed process to set it up, as a *self-managed* environment in HPE GreenLake for Private Cloud Enterprise, for monitoring and alerting applications. Using the instrumentation of multiple supported agents from Apache SkyWalking, the application workloads can be easily monitored through  ...
+
+I﻿n the next blog, we will show you how to use Apache SkyWalking for Kubernetes monitoring in HPE GreenLake for Private Cloud Enterprise.
```
@GuopingJia 

Further to our discussion today and my better understanding of the 2 blog post series, I would suggest the following edit for the last sentence of the Conclusion:

[In the next blog post of the series](https://developer.hpe.com/blog/set-up-apache-skywalking-for-k8s-monitoring-in-hpe-greenlake-for-private-cloud-enterprise/), I will show you how to use Apache SkyWalking APM tool for monitoring of Kubernetes clusters provisioned on HPE GreenLake for Private Cloud Enterprise.

choukden reviewed (changes requested):
Hi Guoping,

Further to our discussion today and my better understanding of the 2 blog post series, I would suggest the following edit for the last sentence of the Conclusion:

[In the next blog post of the series](https://developer.hpe.com/blog/set-up-apache-skywalking-for-k8s-monitoring-in-hpe-greenlake-for-private-cloud-enterprise/), I will show you how to use Apache SkyWalking APM tool for monitoring of Kubernetes clusters provisioned on HPE GreenLake for Private Cloud Enterprise.

GuopingJia commented on content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:
```diff
@@ -0,0 +1,324 @@
+
+After running above commands, t﻿he Apache SkyWalking is installed to the Kubernetes cluster's namespace *skywalking*. It creates the *elasticsearch* as the `StatefulSet`, running a pod on each worker node. It runs the Apache SkyWalking OAP with replicas as 2 to provide high availability. 
+
+It should be noted that the last option *elasticsearch.sysctlInitContainer.enabled=false* is necessary. Otherwise, it will try to set up *vm.max_map_count* using a privileged container during *elasticsearch* installation. It will violate the existing *PodSecurityPolicy* deployed in the HPE GreenLake environment. It therefore will fail the Helm install. 
```
Yes, it's better to say HPE GreenLake for Private Cloud Enterprise environment.

GuopingJia commented on content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:
```diff
@@ -0,0 +1,324 @@
+
+### Deploy a Sample SpringBoot Application
+
+A﻿s the first demo application, create a *SpingBoot* Web app that provides a REST endpoint **/message** to print some nice message. Then generate a *jar* file *springboot-k8s-demo.jar* to the target folder:
```
No problem. I've added the mvn build process with the pom.xml to generate this jar package.

GuopingJia commented on content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:
```diff
@@ -0,0 +1,324 @@
+T﻿he alarms page shows *Successful rate of service agent::app is lower than 80% in 2 minutes of last 10 minutes*. It indicates issue from the frontend app server in multi-tier *music* application.
+
+The alerts are triggered by the following metric alerting rule _service_sla_: 
+
```
Added the process to read the content of the alerting rules.

GuopingJia commented:
Hi, Denis, 

The new patch submitted should address all your comments.

Guoping

drensing commented on content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:
```diff
@@ -0,0 +1,382 @@
+
+### Prerequisites
+
+Before we start, make sure we meet the following requirements:
```
Recommend rephrasing to "Before starting, make sure you have the following requirements:"

drensing commented on content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:
```diff
@@ -0,0 +1,382 @@
+
+Before we start, make sure we meet the following requirements:
+
+* A Kubernetes cluster, being provisioned in HPE GreenLake for Private Cloud Enterprise;
```
Take the punctuation (the semi colons and period) off of the end of each of the following 3 bullets.

drensing commented on content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:
````diff
@@ -0,0 +1,382 @@
+--set elasticsearch.sysctlInitContainer.enabled=false
+```
+
+After running above commands, t﻿he Apache SkyWalking is installed to the Kubernetes cluster's namespace *skywalking*. It creates the *elasticsearch* as the `StatefulSet` object, running a pod on each worker node. It runs the Apache SkyWalking Observability Analysis Platform (OAP) with replicas as 2 to provide high availability. 
````
Add "the" prior to "above commands" , take "the" out prior to "Apache Skywalking", and change "to the Kubernetes" to "on the Kubernetes" so it reads:

"After running the above commands, Apache Skywalking is installed on the Kubernetes..."

It's a little unclear as to what "It" refers to in the next two sentences. I believe that in sentence #2, "It" refers to the Apache SkyWalking app that was just installed. But I'm not sure that's the case in the third sentence. I'm thinking that "It" here refers more to the system you have just installed it on. If that's the case, it would be better to spell out the complete subject at the beginning of the 3rd sentence rather than using the pronoun "It".

drensing replied:
It looks good. Thanks!

drensing commented on content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:
```diff
@@ -0,0 +1,382 @@
+
+After running above commands, t﻿he Apache SkyWalking is installed to the Kubernetes cluster's namespace *skywalking*. It creates the *elasticsearch* as the `StatefulSet` object, running a pod on each worker node. It runs the Apache SkyWalking Observability Analysis Platform (OAP) with replicas as 2 to provide high availability. 
+
+It should be noted that the last option *elasticsearch.sysctlInitContainer.enabled=false* is necessary. Otherwise, it will try to set up *vm.max_map_count* using a privileged container during *elasticsearch* installation. It will violate the existing *PodSecurityPolicy* deployed in the HPE GreenLake for Private Cloud Enterprise environment. It therefore will fail the Helm install. 
```
In this first sentence, do you really mean "option" or should that read "command"?
Be careful to be clear what the pronouns "it" refer to. The first one (at the beginning of the sentence) is fine. But what is "it" that will try to set up *vm.max_map..."? And in the next sentence, it may be more clear to write "By doing so, it will violate the existing.... Enterprise environment and will therefore fail the Helm install."

GuopingJia replied:
Have rephrased those "it". Need your review :-)

drensing replied:
Yup! Rewrite works!

drensing commented on content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:
````diff
@@ -0,0 +1,382 @@
+job.batch/skywalking-es-init   1/1           7m27s      8m6s
+```
+
+You can edit the deployed SkyWalking UI service *skywalking_ui* and change its type from *ClusterIP* to *NodePort*. The service will be automatically mapped to gateway host with an assigned port.
````
Add the word "a" prior to "gateway host" at the end of the second sentence.

drensing commented on content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:
```diff
@@ -0,0 +1,382 @@
+
+### Deploy a sample application: *SpringBoot*
+
+A﻿s the first demo application, create a *SpingBoot* Web app that provides a REST endpoint **/message** to print some nice message. Then generate a *jar* package using the [Apache Maven](https://maven.apache.org/what-is-maven.html) command *mvn* with the *pom.xml* file:
```
Do you really mean "As the first demo application..." or "As **in** the first demo application..." If it's the second, you should rephrase it to "As I had you do in the application I demo'd in my first blog post in this series, create...."

Add a comma after "Then" at the beginning of the next sentence.

drensing commented on content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:
````diff
@@ -0,0 +1,382 @@
+target/springboot-k8s-demo.jar
+```
+
+You can see the *jar* package *springboot-k8s-demo.jar* is built to the *target* folder. B﻿y building a *Docker* image using this generated *jar* file, the *SpringBoot* Web app can be easily deployed as a containerized application to the Kubernetes cluster.
````
I think it would read more clearly if you change "built to the target folder" to "created in the target folder." If you don't care for that, at least change "to" to "in". Similarly, "to the Kubernetes cluster." should be "in the Kubernetes cluster."

drensing commented on content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:
```diff
@@ -0,0 +1,382 @@
+
+You can see the *jar* package *springboot-k8s-demo.jar* is built to the *target* folder. B﻿y building a *Docker* image using this generated *jar* file, the *SpringBoot* Web app can be easily deployed as a containerized application to the Kubernetes cluster.
+
+Apache SkyWalking provides a list of agents for instrumenting applications. A specific agent per programming language can be used for building into corresponding service which collects application data and exports them to the SkyWalking OAP server.   
```
"building into corresponding service" should change to "to build a corresponding service". Also "which" should be changed to "that collects application..."

drensing commented on content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:
```diff
@@ -0,0 +1,382 @@
+
+![](/img/java-agent.png)
+
+H﻿ere is the *Dockerfile* for building the image with Java agent:
```
"with Java agent:" should either read "with the Java agent:" or "using a Java agent:"

drensing commented on content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:
```diff
@@ -0,0 +1,382 @@
+
+### Monitor SpringBoot application from SkyWalking UI
+
+After build the Docker image *guopingjia/springboot-k8s-demo:pce* and push it to the *DockerHub* registry, we deploy the *SpringBoot* Web app to the Kubernetes cluster with the *deployment.yaml* manifest file:
```
Modify first sentence to "After **building** the Docker image *guopingjia/springboot-k8s-demo:pce* and **pushing** it to the *DockerHub* registry, **deploy** the *SpringBoot* **web** app to the Kubernetes cluster with the *deployment.yaml* manifest file:

drensing commented on content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:
```diff
@@ -0,0 +1,382 @@
+
+This blog post took the Apache SkyWalking as the application performance monitoring (APM) tool and showed the detailed process to set it up, as a *self-managed* environment in HPE GreenLake for Private Cloud Enterprise, for monitoring and alerting applications. Using the instrumentation of multiple supported agents from Apache SkyWalking, the application workloads can be easily monitored through  ...
+
+I﻿n [the next blog post of the series](https://developer.hpe.com/blog/set-up-apache-skywalking-for-k8s-monitoring-in-hpe-greenlake-for-private-cloud-enterprise/) , I will show you how to use Apache SkyWalking APM tool for monitoring of Kubernetes clusters provisioned on HPE GreenLake for Private Cloud Enterprise.
```
In this case, you would add the word "the" prior to "Apache SkyWalking APM tool for monitoring...."

drensing reviewed (commented):
Thank you, Guoping! After this one, just one more for me to review.

All the best,
Dale

GuopingJia commented on content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:
```diff
@@ -0,0 +1,382 @@
+
+## Set up Apache SkyWalking for Application Monitoring
+
+I will take the approach to setting up the Apache SkyWalking as a *self-managed* APM tool within the Kubernetes cluster created in HPE GreenLake for Private Cloud Enterprise. This mainly takes into account the security concerns in HPE GreenLake product environment. 
```
You are absolutely right. HPE GreenLake provides excellent security. By setting up the APM tool within this environment, rather than outside, it can benefit from such security.

drensing replied:
Okay, so it should be rewritten to express what you just said, that by setting up the APM tool within this environment, it can benefit from the security features of HPE GreenLake.

GuopingJia replied:
Thanks Dale.  It's been addressed in the new patch.

GuopingJia commented on content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:
```diff
@@ -0,0 +1,382 @@
+
+![](/img/multl-tier-app-music.png)
+
+I﻿n order to monitor this multi-tier *music* application from Apache SkyWalking, you need to pick up the SkyWalking agent per programming language and rebuild corresponding service to instrument the application to the SkyWalking OAP server.
```
Have rephrased it. Please review.

drensing replied:
Oh, yeah... that's much better. Much clearer. Thank you.

GuopingJia commented on content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:
````diff
@@ -0,0 +1,382 @@
+    └── src
+```
+
+A﻿fter image files are rebuilt with the agents, the multi-tier *music* application can be deployed to the Kubernetes cluster:
````
I think you are right. I will go with "in the Kubernetes cluster".

GuopingJia commented:
> Thank you, Guoping! After this one, just one more for me to review.
> 
> All the best, Dale

Thanks Dale for your comments and good suggestions. 

The new patch has addressed all your comments. Let me know if there is still anything we want me to change. 

Guoping

drensing commented:
Re-reviewed the things you asked about. Looks great. I’m green. Thank you.

From: GuopingJia ***@***.***
Sent: Wednesday, January 18, 2023 4:28 AM
To: hpe-dev-incubator/hpe-dev-portal ***@***.***>
Cc: Rensing, Dale ***@***.***>; Review requested ***@***.***>
Subject: Re: [hpe-dev-incubator/hpe-dev-portal] Create Blog “set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud” (PR #1636)


@GuopingJia commented on this pull request.

________________________________

In content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md<https://github.com/hpe-dev-incubator/hpe-dev-portal/pull/1636#discussion_r1073289697>:

> +$ cd skywalking-kubernetes/chart



+$ helm repo add elastic https://helm.elastic.co



+$ helm dep up skywalking



+$﻿ kubectl create ns skywalking



+$ helm install skywalking skywalking –n skywalking \



+--set oap.image.tag=9.2.0 \



+--set oap.storageType=elasticsearch \



+--set ui.image.tag=9.2.0 \



+--set elasticsearch.imageTag=7.17.1 \



+--set elasticsearch.persistence.enabled=true \



+--set elasticsearch.sysctlInitContainer.enabled=false



+```



+



+After running above commands, t﻿he Apache SkyWalking is installed to the Kubernetes cluster's namespace *skywalking*. It creates the *elasticsearch* as the `StatefulSet` object, running a pod on each worker node. It runs the Apache SkyWalking Observability Analysis Platform (OAP) with replicas as 2 to provide high availability.



+



+It should be noted that the last option *elasticsearch.sysctlInitContainer.enabled=false* is necessary. Otherwise, it will try to set up *vm.max_map_count* using a privileged container during *elasticsearch* installation. It will violate the existing *PodSecurityPolicy* deployed in the HPE GreenLake for Private Cloud Enterprise environment. It therefore will fail the Helm install.



Have rephrased those "it". Need your review :-)

—
Reply to this email directly, view it on GitHub<https://github.com/hpe-dev-incubator/hpe-dev-portal/pull/1636#discussion_r1073289697>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ASPH7RIKZBIHDXIM226QJG3WS6ZQLANCNFSM6AAAAAATQUNKWI>.
You are receiving this because your review was requested.Message ID: ***@***.******@***.***>>

GuopingJia commented:
The new patch has addressed the comment from Fred on privileged container policy.
